cassandra/cluster.py

```diff
@@ -2698,9 +2699,11 @@ def _set_result(self, response):
             if self._current_pool and self._connection:
                 self._current_pool.return_connection(self._connection)
 
-            trace_id = getattr(response, 'trace_id', None)
-            if trace_id:
-                self._query_trace = QueryTrace(trace_id, self.session)
+            self._query_trace_id = getattr(response, 'trace_id', None)
```

Comment:
I'm asking because you also assign it to the query, and it looks like that's the instance that gets used in your cqlsh patch.